lineairdb_field.h

```diff
@@ -0,0 +1,55 @@
+#include <string>
+#include "my_inttypes.h"
+
+/**
+ * @brief This class is responsible for the translation between
+ * MySQL Field value and LineairDB Field.
+ * @details
+ * LineairDB field consists of the following 3 information:
+ *  header1     header2
+ * [byteSize][valueLength][value]
+ * header info
+ * - byteSize: number of bytes of `valueLength`
+ *             always 1 byte, byteSize = UCHAR_MAX if valueLength = 0
+ * - valueLength: length of value
+ *                max 8 bytes
+ * value: MySQL value shown to users
+ *        max 4294967295 = sizeof(LONGBLOB) bytes
+ * Each row consists of multiple fields.
+ * First field stores null flags.
+ */
+class Lineairdb_Field
```

Comment:
Do not use a mix of snake_case and CamelCase.
You should follow some coding style guide.
For example, google defines their style guides: https://google.github.io/styleguide/cppguide.html#General_Naming_Rules:

> Type names start with a capital letter and have a capital letter for each new word, with no underscores: MyExcitingClass, MyExcitingEnum.

So if you follows the google's style guide, the naming should be `LineairDBField`.

You can also use a tool called `lint` (e.g., cpplint) to detect that a contributor (including you) violates a style guide and make it fail a test. You should use it if you want to improve the quality of the code of the project.

Reply:
Understood, I will use LineairDBField.